RiotSwiftUI/Modules/UserSessions/UserSessionCardView/UserSessionCardViewData.swift

```diff
@@ -0,0 +1,80 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// View data for UserSessionCardView
+struct UserSessionCardViewData {
+    
+    // MARK: - Constants
+    
+    private static let userSessionNameFormatter = UserSessionNameFormatter()
+    private static let lastActivityDateFormatter = UserSessionLastActivityFormatter()
+        
+    // MARK: - Properties
+    
+    var id: String {
+        return sessionId
+    }
+    
+    let sessionId: String
+
+    let sessionName: String
+    
+    let isVerified: Bool
+    
+    let lastActivityDateString: String?
+    
+    let lastSeenIPInfo: String?
+    
+    let deviceAvatarViewData: DeviceAvatarViewData
+    
+    /// Indicate if the current user session is shown and to adpat the layout
+    let isCurrentSessionDisplayMode: Bool
+    
+    // MARK: - Setup
+    
+    init(sessionId: String,
+         sessionDisplayName: String?,
+         deviceType: DeviceType,
+         isVerified: Bool,
+         lastActivityDate: TimeInterval?,
```

Comment:
It might be a bit clearer if this was named `lastActivityTimestamp` given it isn’t actually a date.